src/audio.c

```diff
@@ -147,7 +152,7 @@ static void audio_callback()
     {
         /* check if next buffer is full */
         int next = (now_playing + 1) % _num_buf;
-        if (!(buf_full & (1<<next)))
+        if ((!(buf_full & (1<<next))) && !_fill_buffer_callback)
```

Comment:
True. I should probably add `&& _fill_buffer_callback` to the first `if` in `audio_pause`.